packages/commonwealth/server/migrations/20240718135145-group-permission-refactor.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    try {
+      await queryInterface.sequelize.transaction(async (transaction) => {
+        // Turn (topic_id, group_id) into the composite primary key
+        await queryInterface.sequelize.query(
+          `
+          ALTER TABLE "GroupPermissions" ADD COLUMN topic_id INTEGER;
+          ALTER TABLE "GroupPermissions" ADD CONSTRAINT GroupPermissions_topic_id_fkey
+          FOREIGN KEY (topic_id) REFERENCES "Topics"(id);
+          ALTER TABLE "GroupPermissions" DROP CONSTRAINT "GroupPermissions_pkey";
+          ALTER TABLE "GroupPermissions" ADD PRIMARY KEY (topic_id, group_id);
+          `,
+          { transaction },
+        );
+
+        const topics = await queryInterface.sequelize.query(
+          `SELECT id, group_ids FROM "Topics" WHERE array_length(group_ids, 1) > 0;`,
+          {
+            type: queryInterface.sequelize.QueryTypes.SELECT,
+            raw: true,
+            transaction,
+          },
+        );
+
+        for (const t of topics) {
+          for (const g of t.group_ids) {
+            // we have consistency issues with the old model.
+            // Some group_ids from topics are referencing non-existing groups, so skip them if we encounter them
+            if (g < 52) {
+              continue;
+            }
```

Comment:
Can we use a constant for this? Magic numbers don't give any context.

Reply:
Good point. I am going to use Tim's query to remove all the inconsistent topic_ids instead.

Reply:
Resolved in [2328c70]